tools/run_tests/performance/prom.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# example usage: python3 prom.py --url=http://10.108.0.60:9090 --pod_type=driver --pod_type=client --container_name=main --container_name=sidecar --debug=true
+import argparse
+import json
+import requests
+import statistics
+import argparse
+from typing import Any, Dict, List
+
+
+class Prom:
+    def __init__(
+            self,
+            url,
+            start,
+            end,
+            debug=False):
+        self.url = url
+        self.start = start
+        self.end = end
+        self.debug = debug
+
+    def fetch_by_query(self, query: str) -> Any:
+        """Fetch the given query with time range"""
+        resp = requests.get(self.url + "/api/v1/query_range", {
+            "query": query,
+            "start": self.start,
+            "end": self.end,
+            "step": 5
+        })
+
+        if not resp.ok:
+            print(resp.content)
+            raise Exception(str(resp))
+
+        return resp.json()
+
+    def fetch_cpu_for_pod(self, container_matcher: str, pod_name: str) -> Dict[str, List[float]]:
+        """Fetch the cpu data for each pod and construct the container
+        name to cpu data list Dict"""
+        query = 'irate(container_cpu_usage_seconds_total{job="kubernetes-cadvisor",pod="' + pod_name + '",container=' + \
+            container_matcher + '}[100s])'
+        if self.debug:
```

Comment:
Maybe try using [a logger](https://docs.python.org/3/howto/logging.html) with levels set according to a flag?